pkg/s3select/select.go

```diff
@@ -175,41 +178,56 @@ func (reader *Input) processSelectReq(reqColNames []string, alias string, whereC
 	if limitOfRecords == 0 {
 		limitOfRecords = math.MaxInt64
 	}
-
 	for {
-		record := reader.ReadRecord()
-		reader.stats.BytesProcessed += processSize(record)
-		if record == nil {
+		individualRecord, errParsing := readIndividualRecord(inputType)
+		if errParsing != nil {
+			rowStruct := &Row{
+				err: errParsing,
+			}
+			myRow <- rowStruct
+			return
+		}
+		out, _ := json.Marshal(individualRecord)
```

Comment:
what if the record was CSV?

Reply:
CSV anf JSON bot are read as map[string]interface so `out, _ := json.Marshal(individualRecord)` is valid for both.

Reply:
I am not able to understand how json.Marshal will be the same as CSV, it is not easy to comprehend this tyle of logic.